roop/core.py

```diff
@@ -193,11 +196,17 @@ def start() -> None:
     update_status('Extracting frames...')
     extract_frames(roop.globals.target_path)
     temp_frame_paths = get_temp_frame_paths(roop.globals.target_path)
-    update_status('Swapping in progress...')
-    conditional_process_video(roop.globals.source_path, temp_frame_paths)
-    # prevent memory leak using ffmpeg with cuda
+    if 'face-swapper' in roop.globals.frame_processors:
+        update_status('Swapping in progress...')
+        conditional_process_video(roop.globals.source_path, temp_frame_paths, roop.swapper.process_video)
     if roop.globals.gpu_vendor == 'nvidia':
         torch.cuda.empty_cache()
+    if 'ROCMExecutionProvider' in roop.globals.providers:
```

Comment:
if this is nvidia/cuda only, then please do remove the ROCM check and just is `if roop.globals.cpu_vendor = 'nvidia'` instead